webapp/src/editortoolbar.tsx

```diff
@@ -208,6 +209,39 @@ export class EditorToolbar extends data.Component<ISettingsProps, EditorToolbarS
         this.props.parent.pairAsync();
     }
 
+    protected onCannotPairClick = async () => {
+        pxt.tickEvent("editortools.pairunsupported", undefined, { interactiveConsent: true });
+        const reasonUnsupported = await pxt.usb.getReasonUnavailable();
+        let modalBody: string;
+        switch (reasonUnsupported) {
+            case "security":
+                modalBody = lf("WebUSB appears to have been disabled by browser policy; check with your technical adminstrator for details.");
+                break;
+            case "oldwindows":
+                modalBody = lf("WebUSB is not available on Windows devices with versions below 8.1.");
+                break;
+            case "electron":
+                modalBody = lf("WebUSB is not supported in electron.");
+                break;
+            case "notimpl":
+                modalBody = lf("WebUSB is not supported by this browser; please check for updates.");
+                break;
+        }
+
+        dialogAsync({
+            header: lf("WebUSB unavailable"),
```

Comment:
I think the header should be "Cannot Connect Device" and then in the body it should say something about WebUSB being unavailable. Otherwise, I'm not sure people will make the connection.